SemanticUI/Modules/Dropdown/Selection.elm

```diff
@@ -0,0 +1,399 @@
+Supports the common use-case where no value is selected and a default placeholder label needs to be supplied.
+-}
+valueLabelWithDefault : Html msg -> (selection -> Html msg) -> Selection msg option (Maybe selection) -> Selection msg option (Maybe selection)
+valueLabelWithDefault defaultHtml valueToHtml =
+    valueLabel (Result.fromMaybe defaultHtml << Maybe.map valueToHtml)
+
+
+{-| A dropdown select component with a single active selection.
+-}
+single :
+    { config
+        | value : selection
+        , isSelected : option -> Bool
+        , drawerState : Drawer.State
+        , identifier : String
+        , onToggle : Drawer.State -> msg
+        , onSelect : option -> msg
+    }
+    -> Selection msg option selection
+single config =
+    let
+        (Select selectConfig) =
+            Select.select
+                { drawerState = config.drawerState
+                , identifier = config.identifier
+                , onToggle = config.onToggle
+                , onSelect = config.onSelect
+                , label = Nothing
+                }
+    in
+    Selection
+        { variation = selectConfig.variation
+        , drawerState = selectConfig.drawerState
+        , identifier = selectConfig.identifier
+        , onToggle = selectConfig.onToggle
+        , onSelect = config.onSelect
+        , attributes = selectConfig.attributes
+        , toggleEvent = selectConfig.toggleEvent
+        , disabled = selectConfig.disabled
+        , dropdownIcon = selectConfig.dropdownIcon
+        , labels = selectConfig.labels
+        , fluid = selectConfig.fluid
+        , scrolling = selectConfig.scrolling
+        , value = config.value
+        , formInput = Nothing
+        , optionAttributes =
+            \option ->
+                [ classList [ ( "active selected", config.isSelected option ) ] ]
+        }
+
+
+{-| A single selection button dropdown that is labeled with the current selection.
+-}
+button :
+    { config
+        | button : Button.Config msg
+        , value : selection
+        , drawerState : Drawer.State
+        , identifier : String
+        , onToggle : Drawer.State -> msg
+        , onSelect : option -> msg
+        , isSelected : option -> Bool
+    }
+    -> Selection msg option selection
+button config =
+    let
+        (Selection singleConfig) =
+            single config
+    in
+    Selection { singleConfig | variation = Select.Button config.button }
+
+
+{-| A single selection dropdown component that is styled similarly to a `<select>` form control.
+
+This configuration can be used with an optional hidden `<input>` for forms.
+
+-}
+selection :
+    { config
+        | value : selection
+        , drawerState : Drawer.State
+        , identifier : String
+        , onToggle : Drawer.State -> msg
+        , onSelect : option -> msg
+        , isSelected : option -> Bool
+        , formInput : Maybe { name : String, toValue : selection -> String }
```

Comment:
`formInput` is the other thing I'm wondering whether we can do without. It seems like it's sort of mostly relevant for traditional form submission, but for modern elm it's all ajax anyway...

It's just that it's what semantic ui does to stay strictly compatible with other form elements I think...